src/utils/re_frame.cljs

```diff
@@ -88,3 +88,37 @@
 (def dispatch-sync re-frame/dispatch-sync)
 
 (def reg-event-fx re-frame/reg-event-fx)
+
+(defn call-continuation
+  "Choose how to call a continuation for a Re-Frame event or effect.
+   
+   When defining an event or effect, we can receive `on-success` and `on-error`
+   parameters for continuing the logic depending on if it succeeded or failed.
+   When we attempt to continue the logic, we can choose to either dispatch a Re-Frame event,
+   or we can call a callback function.
+
+   Code example:
+
+   (rf/reg-event-fx :my-event
+     (fn [_ [arg on-success on-error]]
+       {:fx [[:my-effect [arg on-success on-error]]]}))
+
+   (rf/reg-event-fx :my-event-success
+     (fn [db [result]]
+       {:db (assoc db :my-event-result result)}))
+   
+   (rf/reg-event-fx :my-event-error
+     (fn [db [error]]
+       {:db (assoc db :my-event-error error)}))
+   
+   (rf/reg-fx :my-effect
+     (fn [[arg on-success on-error]]
+       (-> (my-effect-impl arg)
+           (promesa/then (partial call-continuation on-success))
+           (promesa/catch (partial call-continuation on-error)))))
+   
+   (rf/dispatch [:my-event [:stuff] [:my-event-success] [:my-event-error]])"
+  [continuation & args]
+  (cond
+    (vector? continuation) (dispatch (into continuation args))
+    (fn? continuation)     (apply continuation args)))
```

Comment:
Hey @mohsen-ghafouri I've added some documentation with an example for this function. Please let me know what you think 🙏 

cc @ilmotta

Reply:
Thank you @seanstrom. Just two details:

An idea is to change line 120 and show that `:my-event` can take either a function or an event vector.

```clojure
(rf/dispatch [:my-event
              "an arg"
              (fn []
                (log/info "Answer to life: 42")
                (rf/dispatch [:my-event-success]))
              [:my-event-error]])
```

Reply:
Thank you @seanstrom, the structure is significantly more comprehensible now 🙏